test/integration-serial/serial_task_run_test.rb

```diff
@@ -18,7 +18,10 @@ def test_run_without_verify_result_fails_if_pod_was_not_created
     mock.expects(:create_pod).raises(Kubeclient::HttpError.new("409", "Pod with same name exists", {}))
     task_runner.instance_variable_set(:@kubeclient, mock)
 
-    result = task_runner.run(run_params(verify_result: false))
+    result = false
+    metrics = capture_statsd_calls do
+      result = task_runner.run(run_params(verify_result: false))
+    end
```

Comment:
Why in this serialized, stubbed test instead of one of the basic ones in `test/integration/runner_task_test.rb`? Shouldn't we add similar assertions to a timeout and a success case too?

Reply:
`capture_statsd_calls` isn't thread safe

Reply:
I see we're doing it in the main file though, and using the namespace tag to identify metrics that were actually emitted by the test in question (since each test gets its own namespace). Can't we do that for these tests too? The fact that we're lacking coverage in the other suite isn't a good reason not to add good coverage here.

Reply:
Yeah, I misunderstood the point of that comment. 

> The fact that we're lacking coverage in the other suite isn't a good reason not to add good coverage here.

Agree, but given the high quality of the test coverage in the project when I lack context I usually assume these things are intentional. e.g. that the level of coverage was  intentional and reflected that amount of testing needed here.

Reply:
Ok, so you were definitely right that `capture_statsd_calls` _itself_ isn't threadsafe. It's actually [modifying the global statsd configuration](https://github.com/Shopify/statsd-instrument/blob/273017153aa915ac3f02558d1ed691d2a8d68e2f/lib/statsd/instrument/helpers.rb). I'd suggest we make a statsD test in the serial suite (where you had it originally) that covers the three possible results using configurations that succeed/fail/time out as quickly as possible.